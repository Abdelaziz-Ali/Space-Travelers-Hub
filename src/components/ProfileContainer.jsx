export default function ProfileContainer() {
  return (
    <section>
      <div className="row gx-3">
        <div className="col">
          <div className="card">
            ProfileRockets
          </div>
        </div>
        <div className="col">
          <div className="card">
            ProfileMissions
          </div>
        </div>
      </div>
    </section>
  );
}
