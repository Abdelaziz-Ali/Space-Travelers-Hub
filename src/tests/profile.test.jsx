import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import ProfileContainer from '../components/ProfileContainer';
import store from '../redux/store';

describe('Test the profile container components', () => {
  const rocket = (
    <Provider store={store}>
      <ProfileContainer />
    </Provider>
  );

  test('Take the snapshot from the component', () => {
    const component = render(rocket);
    expect(component).toMatchSnapshot();
  });

  test('Profile container has two heading elements', () => {
    render(rocket);
    const headings = screen.getAllByRole('heading');
    expect(headings.length).toBe(2);
  });
});
