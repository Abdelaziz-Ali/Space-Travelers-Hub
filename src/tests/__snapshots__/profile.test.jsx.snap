// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the profile container components Take the snapshot from the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="row gx-3"
        >
          <div
            class="col"
          >
            <h3>
              Reserved Rockets
            </h3>
            <div
              class="alert alert-warning"
              role="alert"
            >
              You have not reserved any rocket yet.
            </div>
          </div>
          <div
            class="col"
          >
            <h3>
              Joined Missions
            </h3>
            <div>
              <div
                class="alert alert-warning"
                role="alert"
              >
                You have not joined to any mission yet.
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="row gx-3"
      >
        <div
          class="col"
        >
          <h3>
            Reserved Rockets
          </h3>
          <div
            class="alert alert-warning"
            role="alert"
          >
            You have not reserved any rocket yet.
          </div>
        </div>
        <div
          class="col"
        >
          <h3>
            Joined Missions
          </h3>
          <div>
            <div
              class="alert alert-warning"
              role="alert"
            >
              You have not joined to any mission yet.
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
